src/datasets/arrow_dataset.py

```diff
@@ -1004,6 +1004,23 @@ def save_to_disk(self, dataset_path: str, fs=None):
             json.dump(sorted_keys_dataset_info, dataset_info_file, indent=2)
         logger.info(f"Dataset saved in {dataset_path}")
 
+    @staticmethod
+    def build_local_temp_path(uri_or_path: str) -> Path:
```

Comment:
Could it be a private method ?
```suggestion
    def _build_local_temp_path(uri_or_path: str) -> Path:
```

Reply:
Sure! It should not be exposed.